lib/dgram.js

```diff
@@ -272,6 +267,65 @@ Socket.prototype.send = function(buffer,
   if (offset + length > buffer.length)
     throw new RangeError('Offset + length beyond buffer length');
 
+  return buffer.slice(offset, offset + length);
+}
+
+function fixBuffer(buffer) {
+  var length = 0;
+
+  for (var i = 0, l = buffer.length; i < l; i++) {
+    if (typeof buffer[i] === 'string')
+      buffer[i] = new Buffer(buffer[i]);
```

Comment:
Optimal case you'd allow the user to pass a string to C++, copy out the memory there and store a pointer to it on the class. Then free the memory in the destructor. This would have to be a linked list, since you're accepting an array of values, but would remove the additional overhead of creating another Buffer object and the GC cost of cleaning it up.